### PR TITLE
Minor change to test how this affects the rest of CI

### PR DESCRIPTION
**Pull request by costrouc**
Closes https://github.com/conda/conda-libmamba-solver/issues/156
Closes https://github.com/conda/conda-libmamba-solver/issues/71

### Description

This PR aims to fix issues when a user run `conda update ...` and in turn conda-libmamba-solver recommends to downgrade the users packages. This PR involves investigating the spec <-> tasks translations and how these are fed into libsolv.

### Checklist - did you ...

- [ ] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

**Comment by costrouc** [comment]
I was able to demonstrate how setting `>=...` did fix the issue shown in #71 but I'm curious how this affects other tests.

**Comment by costrouc** [comment]
Current failing tests:
 - tests/test_create.py::IntegrationTests::test_create_only_deps_flag https://github.com/conda/conda-libmamba-solver/actions/runs/4496472144/jobs/7911184496?pr=158#step:5:1611 (several times)
 - tests/gateways/test_connection.py::test_s3_server https://github.com/conda/conda-libmamba-solver/actions/runs/4496472144/jobs/7911181839?pr=158#step:9:1075 (several times)

I can reproduce the first one but on the s3 one.

File: conda_libmamba_solver/solver.py [hunk 1]
```diff
@@ -453,6 +453,10 @@ def _specs_to_tasks_add(self, in_state: SolverInputState, out_state: SolverOutpu
                             "UPDATE | ESSENTIAL | FORCEBEST",
                             api.SOLVER_UPDATE | api.SOLVER_ESSENTIAL | api.SOLVER_FORCEBEST,
                         )
+                    elif self._command == "update":
```
Comment (jaimergp) [inline conda_libmamba_solver/solver.py:456]: Hm, we already checked for this in line 445 above. I wonder if we can simply change the `!=` in the else below? I remember experimenting with this but something else broke. And we also need to check for the error messages. If a package cannot be updated because the latest version is already there, then that's not an error per se.

**Comment by jaimergp** [comment]
> I can reproduce the first one but not the s3 one.

The S3 one is unrelated. I've seen it in other PRs. I think `minio` released a new server or something and we need to pin.

**Comment by costrouc** [comment]
Link to tests on d6c66ca
 - https://github.com/conda/conda-libmamba-solver/actions/runs/4673486979
 - https://github.com/conda/conda-libmamba-solver/actions/runs/4673486981

**Comment by jaimergp** [comment]
Now we have some more clarity on the failing tests:

```
FAILED ../conda-libmamba-solver-src/tests/test_modified_upstream.py::test_freeze_deps_1
FAILED ../conda-libmamba-solver-src/tests/test_modified_upstream.py::test_conda_downgrade
```

These are upstream tests that we vendored to modify the expectations a bit. There's a chance we can readjust them again, but also it might be that the `>=` changes introduce this regression too. To be investigated.

cc @costrouc

**Comment by costrouc** [comment]
@jaimergp getting on this right now. Thanks for rebasing to rerun the CI.

**Comment by jaimergp** [comment]
So this might be working now! The only thing that changed is that libmamba chooses to keep one dependency around (`xz`) while conda classic prunes it. In the past we could workaround that in the test by adding `prune=True` to the solver, but not anymore. So I removed that modification and commented out the `xz` line. It's sufficiently edge-casey (and keeping a package is not too critical of an error) so we can ignore it, imho.

File: conda_libmamba_solver/solver.py [hunk 2]
```diff
@@ -262,7 +262,10 @@ def _solving_loop(
                 "as conflicts for maximum flexibility",
             )
             # we only check this for "desperate" strategies in _specs_to_tasks
-            self._command = "last_solve_attempt"
+            if self._command in (None, NULL):
```
Comment (jezdez) [inline conda_libmamba_solver/solver.py:265]: zilch, nada!

**Review by jezdez — APPROVED** [review]
